database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java

```diff
@@ -79,14 +81,14 @@ public AbstractDBSource(DBSourceConfig sourceConfig) {
   }
 
   private static String removeConditionsClause(String importQueryString) {
-    importQueryString = importQueryString.replaceAll("\\s{2,}", " ").toUpperCase();
-    if (importQueryString.contains("WHERE $CONDITIONS AND")) {
-      importQueryString = importQueryString.replace("$CONDITIONS AND", "");
-    } else if (importQueryString.contains("WHERE $CONDITIONS")) {
-      importQueryString = importQueryString.replace("WHERE $CONDITIONS", "");
-    } else if (importQueryString.contains("AND $CONDITIONS")) {
-      importQueryString = importQueryString.replace("AND $CONDITIONS", "");
-    } else if (importQueryString.contains("$CONDITIONS")) {
+    importQueryString = importQueryString.replaceAll("\\s{2,}", " ");
+    if (importQueryString.toUpperCase().contains("WHERE $CONDITIONS AND")) {
+      importQueryString = importQueryString.replaceAll("(?i)" + Pattern.quote("$CONDITIONS AND"), "");
```

Comment:
can you add a comment about what this is doing and also include an example?

Reply:
There was a bug related to queries case insensitiveness. I was not able to specify lowercase table names for MySQL Source. Thus, I removed `toUpperString` conversion for the import query and changed the following `replace` methods to ignore case, since they were expecting string in upper case. This method was added to exclude conditions clause from the query in get schema request in DbSource(https://github.com/cdapio/hydrator-plugins/pull/528).